Sources/PackageCollectionsSigning/PackageCollectionSigning.swift

```diff
@@ -15,7 +15,42 @@ import Basics
 import PackageCollectionsModel
 import TSCBasic
 
-public struct PackageCollectionSigning {
+public protocol PackageCollectionSigner {
+    /// Signs package collection using the given certificate and key.
+    ///
+    /// - Parameters:
+    ///   - collection: The package collection to be signed
+    ///   - certChainPaths: Paths to all DER-encoded certificates in the chain. The certificate used for signing
+    ///                     must be the first in the array.
+    ///   - certPolicyKey: The key of the `CertificatePolicy` to use for validating certificates
+    ///   - certPrivateKeyPath: Path to the private key (*.pem) of the certificate
+    ///   - jsonEncoder: The `JSONEncoder` to use
+    ///   - callback: The callback to invoke when the signed collection is available.
+    func sign(collection: PackageCollectionModel.V1.Collection,
+              certChainPaths: [URL],
+              certPrivateKeyPath: URL,
+              certPolicyKey: CertificatePolicyKey,
+              jsonEncoder: JSONEncoder,
```

Comment:
question: are we passing a concrete JSONEncoder to save the need to create a new one every time?

Reply:
> are we passing a concrete JSONEncoder to save the need to create a new one every time?

No, the collection is serialized and embedded inside the signature, and upon checking the signature we deserialize the collection and compare it with the one received. In previous implementations we were comparing JSON/`Data` so we had to make sure we use the same encoder/decoder, but now that we are comparing `SignedCollection` we can probably get away with that.